modules/rte_mis_core/src/Controller/LogsDownload.php

```diff
@@ -148,4 +148,83 @@ public function getLocationLogs(int $fid = 0) {
     return new Response('File not found!', Response::HTTP_NOT_FOUND);
   }
 
+  /**
+   * Export the logs from student performance import.
+   */
+  public function getStudentPerformanceLogs(int $fid = 0) {
+    $file = $this->entityTypeManager()->getStorage('file')->load($fid);
+    if ($file instanceof FileInterface) {
+      if ($file->getOwnerId() !== $this->currentUser()->id()) {
+        throw new AccessDeniedHttpException('Cannot access the log.');
+      }
+      $destinationUri = 'public://student-performance-logs';
+      $this->fileSystem->prepareDirectory($destinationUri, FileSystemInterface:: CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS);
+      $newFile = $this->fileRepository->copy($file, $destinationUri, FileSystemInterface::EXISTS_RENAME);
+      if ($newFile instanceof FileInterface) {
+        $newFileUri = $this->fileSystem->realpath($newFile->getFileUri());
+        $spreadsheet = IOFactory::load($newFileUri);
+        // Make changes to the spreadsheet.
+        $sheet = $spreadsheet->getActiveSheet();
+        // Get the store collection.
+        $store = $this->tempStoreFactory->get('rte_mis_student_tracking');
+        $studentPerformanceLogs = $store->get('student_performance_logs');
+        if (!empty($studentPerformanceLogs)) {
+          $sheet->getStyle('A1:R1')->applyFromArray([
+            'font' => [
+              'bold' => TRUE,
```

Comment:
Looks like this code is getting repeated.

Reply:
Right removing this.